File: benchmark-java-sync/src/main/java/com/example/benchmarkjavasync/api/ApplicationController.java
package com.example.benchmarkjavasync.api;

import com.example.benchmarkjavasync.core.model.ApplicationResponse;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.Optional;

@RestController
@RequiredArgsConstructor
@Slf4j
public class ApplicationController {

    private final ServiceAdapter applicationService;

    @PostMapping("users/create")
    public Optional<ApplicationResponse> postUser(@RequestBody ApplicationRequestDto userRequestDto) {
        return applicationService.process(userRequestDto);
    }
}

File: benchmark-kotlin-async/src/main/kotlin/com/example/benchmarkkotlinasync/api/ApplicationController.kt
package com.example.benchmarkkotlinasync.api

import com.example.benchmarkkotlinasync.core.model.ApplicationResponse
import lombok.RequiredArgsConstructor
import lombok.extern.slf4j.Slf4j
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
@RequiredArgsConstructor
@Slf4j
class ApplicationController(private val applicationService: ServiceAdapter) {

    @PostMapping("users/create")
    suspend fun postUser(@RequestBody userRequestDto: ApplicationRequestDto): ApplicationResponse {
        return applicationService.process(userRequestDto)
    }
}
